daprdocs/content/en/operations/components/pluggable-components-registration.md

```diff
@@ -49,49 +49,43 @@ Both your component and the Unix Socket must be running before Dapr starts.
 
 By default, Dapr looks for [Unix Domain Socket][uds] files in the folder in `/tmp/dapr-components-sockets`.
 
-Filenames in this folder are significant for component registration. They must be formed by appending the component's name with a file extension of your choice, more commonly `.sock`. For example, the filename `my-component.sock` is a valid UDS file name for a component named `my-component`.
+Filenames in this folder are significant for component registration. They must be formed by appending the component's **name** with a file extension of your choice, more commonly `.sock`. For example, the filename `my-component.sock` is a valid UDS file name for a component named `my-component`.
 
 Since you are running Dapr in the same host as the component, verify this folder and the files within it are accessible and writable by both your component and Dapr.
 
-## Define the component
+### Building-blocks discovery from  behind the same component
 
-Define your component using a [component spec]({{< ref component-schema.md >}}). Your component's `type` is derived from the socket name, without the file extension.
+A pluggable component accessible through a [Unix Domain Socket][uds] can host multiple distinct bulding blocks. During the components initial discovery process, Dapr will use reflection to enumerate all building blocks behind a UDS. The `my-component` pluggable component from the example above could contain both a state store (`state`) and a Pub/Sub (`pubsub`) building-blocks.
 
-Save the component YAML file in the resources-path, replacing:
+The most common use-case is for a 1:1 mapping between pluggable components and building blocks they expose. That said, having the possibility of consolidating multiple building blocks under the same pluggable components might be healpful, specially to ease DevOps burden at the expense of fault tolerance and security. Please weight the pros and cons and, if in doubt, stick to a 1:1 approach.
 
-- `your_socket_goes_here` with your component socket name (no extension)
-- `your_component_type` with your component type
 
-```yaml
-apiVersion: dapr.io/v1alpha1
-kind: Component
-metadata:
-  name: prod-mystore
-spec:
-  type: your_component_type.your_socket_goes_here
-  version: v1
-  metadata:
-```
+## Define the component
 
-Using the previous `my-component.sock` example:
+Define your component using a [component spec]({{< ref component-schema.md >}}). Your component's `spec.type` value is made by concatenating the following 2 parts with a `.`:
+1. the component's **building block type** (`state`, `pubsub` etc)
```

Comment:
Might we be more explicit and enumerate all the possible types?

Reply:
We only have 1 more (bindings)  for the time being but this would be another list to update as we add support to more building blocks/components. The "etc" was a deliberate choice to avoid this getting out of sync 😅

Reply:
Yes, I know it's a maintenance burden, but sometimes having concrete examples (for confusing configuration) can be helpful rather than require user to interpolate.

Reply:
I can add a link to this table here https://docs.dapr.io/operations/components/pluggable-components/pluggable-components-overview/#implementing-a-pluggable-component.

To add to this confusiong: on that page it uses "Component API" for what I am using "building block" for.

Reply:
Building Block is the wrong term here, which are these Dapr APIs https://docs.dapr.io/reference/api/
Pluggable components implement component APIs or interfaces. For example this https://github.com/dapr/components-contrib/blob/a494f3129d7c151131850cbfe72cf11e4379b95e/state/store.go#L24